ports/geckolib/gecko_bindings/sugar/refptr.rs

```diff
@@ -0,0 +1,53 @@
+use std::marker::PhantomData;
+use std::mem::transmute;
+use std::ptr::null;
+use std::sync::Arc;
+
+pub unsafe trait RefCounted<'a> where Self: Sized {
+    type Strong;
+    type Borrowed;
+    fn into_strong(x: Arc<Self>) -> Self::Strong;
+    fn null_strong() -> Self::Strong;
+    fn from_borrowed(x: Self::Borrowed) -> &'a Arc<Self>;
```

Comment:
I understand that right now the objects which we would want to handle with this system are all Rust refcounted objects (Arc<T>s) which we want to pass to Gecko, but what if we have a Gecko refcounted object (RefPtr<T>) which we want to pass to Servo? We won't be able to implement a from_borrowed in that case.